components/services/brave_wallet/zcash/zcash_decoder_unittest.cc

```diff
@@ -0,0 +1,195 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/services/brave_wallet/zcash/zcash_decoder.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/test/bind.h"
+#include "brave/components/brave_wallet/common/proto/protobuf_utils.h"
+#include "brave/components/brave_wallet/common/proto/zcash_grpc_data.pb.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+namespace {
+std::vector<uint8_t> ToBytes(const std::string& str) {
+  return std::vector<uint8_t>(str.begin(), str.end());
+}
+}  // namespace
+
+class ZCashDecoderUnitTest : public testing::Test {
+ public:
+  ZCashDecoderUnitTest() = default;
+  ZCashDecoderUnitTest(const ZCashDecoderUnitTest&) = delete;
+  ZCashDecoderUnitTest& operator=(const ZCashDecoderUnitTest&) = delete;
+  ~ZCashDecoderUnitTest() override = default;
+
+  void SetUp() override { decoder_ = std::make_unique<ZCashDecoder>(); }
+
+  void TearDown() override {}
+
+  ZCashDecoder* decoder() { return decoder_.get(); }
+
+ private:
+  std::unique_ptr<ZCashDecoder> decoder_;
+};
+
+TEST_F(ZCashDecoderUnitTest, ParseBlockID) {
+  // Correct
+  {
+    zcash::BlockID block_id;
+    block_id.set_height(64u);
+    block_id.set_hash("abcd");
+    mojom::BlockIDPtr result;
+    auto callback = base::BindLambdaForTesting(
+        [&result](mojom::BlockIDPtr value,
+                  const absl::optional<std::string>& error) {
+          result = std::move(value);
+        });
```

Comment:
Prefer base::MockCallback + EXPECT_CALL to lambdas

Reply:
Done